compiler/cpp/src/generate/t_java_generator.cc

```diff
@@ -3967,14 +3972,42 @@ string t_java_generator::function_signature_async(t_function* tfunction, bool us
  * @param tfunction Function definition
  * @return String of rendered function definition
  */
-string t_java_generator::function_signature_future(t_function* tfunction, string prefix) {
+string t_java_generator::function_signature_future(t_function* tfunction, int params_futures_mask, string prefix) {
   t_type* ttype = tfunction->get_returntype();
   std::string return_type = type_name(ttype, true);
   std::string fn_name = get_rpc_method_name(tfunction->get_name());
-  std::string arg_list = argument_list(tfunction->get_arglist());
+  std::string arg_list = futures_argument_list(tfunction->get_arglist(), params_futures_mask);
   return "CompletableFuture<" + return_type + "> " + prefix + fn_name + "(" + arg_list + ")";
 }
 
+/**
+ * Renders a comma separated field list, with type names. The types are
+ * modified to use a future of the argument type according to the mask
+ * given.
+ */
+string t_java_generator::futures_argument_list(t_struct* tstruct, int params_futures_mask) {
```

Comment:
This was on purpose. Since I don't know our plans for this library, I tried to keep all the medallia code separated as much as possible, to facilitate future merges in following versions. If the plan is to merge this to the original thrift library, then your suggestions makes sense.